dartagnan/src/main/java/com/dat3m/dartagnan/program/processing/CallStackAnalysis.java

```diff
@@ -0,0 +1,68 @@
+package com.dat3m.dartagnan.program.processing;
+
+import com.dat3m.dartagnan.program.Program;
+import com.dat3m.dartagnan.program.Thread;
+import com.dat3m.dartagnan.program.event.core.*;
+import com.dat3m.dartagnan.program.event.core.annotations.FunCall;
+import com.dat3m.dartagnan.program.event.core.annotations.FunRet;
+import org.sosy_lab.common.configuration.Configuration;
+import org.sosy_lab.common.configuration.InvalidConfigurationException;
+
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Stack;
+
+public class CallStackAnalysis  implements ProgramProcessor {
```

Comment:
We usually do not inherit from `ProgramProcessor` for analyses, but only for transformation code.